### PR TITLE
Add <Avatar> to display an image along with the text

### PR DESCRIPTION
**Pull request by garfieldduck**
# Purpose

In our project in development, there is a need to add users' avatars alongside the text of `<ListRow>`, `<SelectRow>`, and `<SelectOption>`.

With this in mind, we add the following three types (square, rounded, and circle) of `<Avatar>`s to display an image along with the text, and change related components to accommodate the avatars.

Screen Shot 2019-04-26 at 6 24 45 PM

# Screenshots

#### `<Avatar>`
Screen Shot 2019-04-26 at 6 27 39 PM

#### `<ListRow>`
Screen Shot 2019-04-26 at 6 29 26 PM

### `<Checkbox>`
Screen Shot 2019-04-26 at 6 30 27 PM

### `<SelectRow>`
Kapture 2019-04-26 at 18 32 54

# Changes

- Add `<Avatar>`
- Change `rowComp()` to allow the appearance of `<Avatar>` alongside the text 
- Change `<SelectRow>` and `<Checkbox>` to display `<Avatar>`

# Risks

The change of `rowComp()` allows any components wrapped with it to integrate `<Avatar>` on its left. However, since `<Avatar>` is an experimental feature in a project, components except`<ListRow>` and `<SelectRow>` have not been tested and documented.

### CODE REVIEW

File: packages/core/src/styles/Avatar.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    &--circle {
+        border-radius: 50%;
+    }
+
+    // ----------------------
+    //  Inner image
+    // ----------------------
+    > img {
```
Comment (zhusee2) [inline packages/core/src/styles/Avatar.scss]: You probably need a `&` here?
```suggestion
    & > img {
```

File: packages/form/src/SelectRow.js [hunk 2]
```diff
@@ -50,6 +50,22 @@ function getValueLabelMap(fromChildren = []) {
     return resultMap;
 }
 
+/**
+ * Generate a value-avatar map from all `<SelectOption>`s.
+ *
+ * @param {array} fromOptions
+ * @return {Map}
+ */
+function getValueAvatarMap(fromChildren = []) {
```
Comment (zhusee2) [inline packages/form/src/SelectRow.js]: Can this be merged with `getValueLabelMap` so we don't need to iterate the Array twice, only for deriving different data?

File: packages/storybook/examples/core/Checkbox/BasicCheckbox.js [hunk 3]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
 
 import Checkbox from '@ichef/gypcrete/src/Checkbox';
+import Avatar from '@ichef/gypcrete/src/Avatar';
 import DebugBox from 'utils/DebugBox';
 
 function BasicCheckboxExample() {
+    const rdAvatar = <Avatar type="square" alt="Avatar of RD" src="https://api.adorable.io/avatars/285/rd@ichef.tw" />;
```
Comment (zhusee2) [inline packages/storybook/examples/core/Checkbox/BasicCheckbox.js]: Can we change the email here to more generic one, maybe `johndoe@example.com`?

File: packages/form/src/SelectOption.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ export const TYPE_SYMBOL = Symbol('SelectOption');
 function SelectOption({
     label,
     value,
+    avatar,
```
Comment (zhusee2) [inline packages/form/src/SelectOption.js]: missing type check here?

File: packages/form/src/SelectRow.js (hunk 2 above)
Comment (garfieldduck) [reply packages/form/src/SelectRow.js]: Nice suggestion. Thanks!

**Comment by garfieldduck** [comment]
@zhusee2 The code has been refined. You may take a look at it when you're free 😃

**Pull request merged**